### PR TITLE
add gameserver values as configurable in helm terraform modules

### PR DESCRIPTION
**Pull request by comerford**
**What type of PR is this?**

/kind feature

**What this PR does / Why we need it**:

As described in [Issue 1692](https://github.com/googleforgames/agones/issues/1692) - this PR aims to add the gameserver variables (port range, namespaces) to the terraform module

**Which issue(s) this PR fixes**:

https://github.com/googleforgames/agones/issues/1692

Closes #

**Special notes for your reviewer**:

### CODE REVIEW

File: install/terraform/modules/helm/helm.tf [hunk 1]
```diff
@@ -153,6 +153,26 @@ resource "helm_release" "agones" {
     name  = "agones.featureGates"
     value = var.feature_gates
   }
+  
+  set {
```
Comment (aLekSer) [inline install/terraform/modules/helm/helm.tf:157]: This is a duplicate of lines 152:155
```suggestion
```

File: install/terraform/modules/helm/helm.tf [hunk 2]
```diff
@@ -153,6 +153,26 @@ resource "helm_release" "agones" {
...
+    value = var.feature_gates
+  }
+
+  set {
+    name  = "gameservers.namespaces"
+    value = var.gameserver_namespaces
+  }
+
+  set {
+    name  = "gameservers.minport"
```
Comment (aLekSer) [inline install/terraform/modules/helm/helm.tf:168]: Should it be the same as in `values.yaml`?
```suggestion
    name  = "gameservers.minPort"
```

File: install/terraform/modules/helm/variables.tf [hunk 3]
```diff
@@ -72,3 +72,15 @@ variable "ping_service_type" {
...
+  default = "7000"
+}
+
+variable "gameserver_maxport" {
+  default = "8000"
+}
+
+variable "gameserver_namespaces" {
+  default = ["default"]
+}
```
Comment (aLekSer) [inline install/terraform/modules/helm/variables.tf:86]: nit: EOL

File: install/terraform/modules/helm3/helm.tf [hunk 4]
```diff
@@ -90,4 +90,25 @@ resource "helm_release" "agones" {
     name  = "agones.featureGates"
     value = var.feature_gates
   }
+
+  set {
```
Comment (aLekSer) [inline install/terraform/modules/helm3/helm.tf:94]: Same dup:
```suggestion
```

File: install/terraform/modules/helm3/helm.tf [hunk 5]
```diff
@@ -90,4 +90,25 @@ resource "helm_release" "agones" {
...
+    value = var.feature_gates
+  }
+
+  set {
+    name  = "gameservers.namespaces"
+    value = var.gameserver_namespaces
+  }
+
+  set {
+    name  = "gameservers.minport"
```
Comment (aLekSer) [inline install/terraform/modules/helm3/helm.tf:105]: Same as above, should be Camel case

File: install/terraform/modules/helm/helm.tf [hunk 6]
```diff
@@ -153,6 +153,25 @@ resource "helm_release" "agones" {
     name  = "agones.featureGates"
     value = var.feature_gates
   }
+  
+    name  = "agones.featureGates"
+    value = var.feature_gates
+  }
```
Comment (aLekSer) [inline install/terraform/modules/helm/helm.tf:159]: Hello @comerford I mean remove all lines which are duplicate with `featureGates`
```suggestion
```
Sorry, I was not able to make multiline suggestion from the first attempt
https://github.blog/changelog/2020-02-26-multi-line-code-suggestions-beta/

File: install/terraform/modules/helm3/helm.tf [hunk 7]
```diff
@@ -90,4 +90,24 @@ resource "helm_release" "agones" {
     name  = "agones.featureGates"
     value = var.feature_gates
   }
-}
\ No newline at end of file
+
+    name  = "agones.featureGates"
+    value = var.feature_gates
+  }
+
```
Comment (aLekSer) [inline install/terraform/modules/helm3/helm.tf:97]: Same as above
```suggestion
```

File: install/terraform/modules/helm/helm.tf [hunk 8]
```diff
@@ -153,6 +153,25 @@ resource "helm_release" "agones" {
...
+    value = var.gameserver_namespaces
+  }
+
+  set {
+    name  = "gameservers.minPort"
+    value = var.gameserver_minport
+  }
+
+  set {
+    name  = "gameservers.maxport"
```
Comment (aLekSer) [inline install/terraform/modules/helm/helm.tf:172]:
```suggestion
    name  = "gameservers.maxPort"
```

File: install/terraform/modules/helm3/helm.tf [hunk 9]
```diff
@@ -90,4 +90,24 @@ resource "helm_release" "agones" {
...
+  set {
+    name  = "gameservers.namespaces"
+    value = var.gameserver_namespaces
+  }
+
+  set {
+    name  = "gameservers.minport"
```
Comment (aLekSer) [inline install/terraform/modules/helm3/helm.tf:104]:
```suggestion
    name  = "gameservers.minPort"
```

File: install/terraform/modules/helm3/helm.tf [hunk 10]
```diff
@@ -90,4 +90,24 @@ resource "helm_release" "agones" {
...
+    value = var.gameserver_minport
+  }
+
+  set {
+    name  = "gameservers.maxport"
```
Comment (aLekSer) [inline install/terraform/modules/helm3/helm.tf:109]:
```suggestion
    name  = "gameservers.maxPort"
```

**Comment by aLekSer** [comment]
Overall is much better. But please include these 3 new parameters into a website by updating https://github.com/googleforgames/agones/blob/54cc643018841b37377ffe3415808c5a71daaa6d/site/content/en/docs/Installation/Terraform/gke.md#L92  https://agones.dev/site/docs/installation/terraform/gke/#installation
Thank you

**Comment by comerford** [comment]
Added the options in the three docs, I think that is all of it, but if I missed something let me know

File: site/content/en/docs/Installation/Terraform/aks.md [hunk 11]
```diff
@@ -50,6 +50,9 @@ Configurable parameters:
 - region - the location of the cluster
 - node_count - count of game server nodes for the default node pool (default is "4")
 - feature_gates - a list of alpha and beta version features to enable. For example, "PlayerTracking=true&ContainerPortAllocation=true"
+- gameserver_minPort - the lower bound of the port range which gameservers will listen on (default is "7000")
```
Comment (roberthbailey) [inline site/content/en/docs/Installation/Terraform/aks.md:53]: These should be version gated on the next agones release, right?

File: install/terraform/modules/helm/helm.tf [hunk 12]
```diff
@@ -153,6 +153,22 @@ resource "helm_release" "agones" {
     name  = "agones.featureGates"
     value = var.feature_gates
   }
+
```
Comment (roberthbailey) [inline install/terraform/modules/helm/helm.tf:156]: nit: extra blank line.

**Comment by markmandel** [comment]
Just confirming - this looks like it's waiting on a couple of small tweaks from @roberthbailey - is that correct?

File: site/content/en/docs/Installation/Terraform/aks.md (hunk 11 above)
Comment (markmandel) [reply site/content/en/docs/Installation/Terraform/aks.md:53]: https://agones.dev/site/docs/contribute/#within-a-page guide on feature gating within a page.

**Comment by roberthbailey** [comment]
> this looks like it's waiting on a couple of small tweaks from @roberthbailey - is that correct?

If you mean waiting on @comerford making small tweaks to address my feedback, then yes, that is my understanding.

**Comment by comerford** [comment]
Hey all - apologies for lack of action, I am on vacation at the moment, so having to make edits in the Github web interface rather than from a command line. I think I addressed the minor tweaks requested.

**Pull request merged**